tests/unit/s2n_resume_test.c

```diff
@@ -1442,6 +1415,71 @@ int main(int argc, char **argv)
         };
 
+        /* Check error is thrown when wrong version number is read */
+        {
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(conn);
+            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(config);
+
+            EXPECT_OK(s2n_resumption_test_ticket_key_setup(config));
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+            conn->actual_protocol_version = S2N_TLS12;
+            conn->handshake.handshake_type = NEGOTIATED;
+
+            EXPECT_OK(s2n_resume_encrypt_session_ticket(conn, &conn->client_ticket_to_decrypt));
+            EXPECT_NOT_EQUAL(s2n_stuffer_data_available(&conn->client_ticket_to_decrypt), 0);
+
+            /* Modify the version number of the ticket */
+            uint8_t *version_num = conn->client_ticket_to_decrypt.blob.data;
+            EXPECT_EQUAL(*version_num, S2N_PRE_ENCRYPTED_STATE_V1);
+            *version_num = S2N_PRE_ENCRYPTED_STATE_V1 + 100;
+
+            EXPECT_ERROR_WITH_ERRNO(s2n_resume_decrypt_session_ticket(conn, &conn->client_ticket_to_decrypt),
+                    S2N_ERR_SAFETY);
+
+            /* The correct version number should succeed */
+            *version_num = S2N_PRE_ENCRYPTED_STATE_V1;
+            EXPECT_SUCCESS(s2n_stuffer_reread(&conn->client_ticket_to_decrypt));
+            EXPECT_OK(s2n_resume_decrypt_session_ticket(conn, &conn->client_ticket_to_decrypt));
+        }
+
+        /* Check error is thrown when info bytes used to generate the ticket key are incorrect */
+        {
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(conn);
+            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(config);
+
+            EXPECT_OK(s2n_resumption_test_ticket_key_setup(config));
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+            conn->actual_protocol_version = S2N_TLS12;
+            conn->handshake.handshake_type = NEGOTIATED;
+
+            DEFER_CLEANUP(struct s2n_stuffer valid_ticket = { 0 }, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&valid_ticket, 0));
+            EXPECT_OK(s2n_resume_encrypt_session_ticket(conn, &valid_ticket));
+            uint32_t ticket_size = s2n_stuffer_data_available(&valid_ticket);
+
+            /* Copy ticket so that we don't modify the original ticket */
+            EXPECT_SUCCESS(s2n_stuffer_copy(&valid_ticket, &conn->client_ticket_to_decrypt,
+                    ticket_size));
+
+            /* Zero out the info bytes on the ticket */
+            uint8_t *info_ptr = conn->client_ticket_to_decrypt.blob.data + S2N_TICKET_VERSION_SIZE
+                    + S2N_TICKET_KEY_NAME_LEN;
```

Comment:
This index could change if the ticket format changes. Would it be reasonable to read the version and assert v1 here to make sure the test doesn't start memsetting something random if this happens? Or maybe there's another way to ensure this?

Reply:
Yeah, we can't really assert random bytes, but we can read up to the info bytes and assert that they are what is expected.